File: app/src/main/java/com/nollpointer/dates/practise/PractiseCardsAdapter.java
package com.nollpointer.dates.practise;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.nollpointer.dates.R;

import java.util.Random;

public class PractiseCardsAdapter extends RecyclerView.Adapter<PractiseCardsAdapter.ViewHolder> {
    private String[] title_texts;
    private String[] subtitle_texts;
    private int[] imageIds;
    private int[] backgrounds;
    private int[] marks;

    private Listener mListener;

    private int currentMode = 0;

    public interface Listener {
        void onClick(int position);
    }

    public void setListener(Listener listener) {
        mListener = listener;
    }

    public PractiseCardsAdapter(String[] title, String[] subtitle, int[] images, int[] backgrounds) {
        title_texts = title;
        subtitle_texts = subtitle;
        this.imageIds = images;
        this.backgrounds = backgrounds;
    }

    public void setMode(int practiseMode) {
        currentMode = practiseMode;
    }

    public void setMarks(int[] marks){
        this.marks = marks;
    }

    @Override
    public PractiseCardsAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.card_practise, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, final int position) {
        View cardView = holder.itemView;

        TextView titleTextView = cardView.findViewById(R.id.info_text_title);
        TextView subTitleTextView = cardView.findViewById(R.id.info_text_subtitle);
        ImageView markImageView = cardView.findViewById(R.id.info_image_mark);
        ImageView imageView = cardView.findViewById(R.id.info_image);

        titleTextView.setText(title_texts[position]);
        subTitleTextView.setText(subtitle_texts[position]);
        imageView.setImageResource(imageIds[position]);
        imageView.setBackgroundResource(backgrounds[position]);
        imageView.setContentDescription(title_texts[position]);

        if (currentMode == 1) {
            markImageView.setVisibility(View.VISIBLE);
            markImageView.setImageResource(getMarkImage(marks[position]));
        }
    }

    @Override
    public int getItemCount() {
        return imageIds.length;
    }

    private int getMarkImage(int mark) {
        //int colors[] = {0xFFB71C1C,0xFFFFEB3B, 0xFF43a047};
        int colors[] = {R.drawable.ic_practise_result_nothing,R.drawable.ic_practise_result_bad, R.drawable.ic_practise_result_neutral, R.drawable.ic_practise_result_good};

        return colors[mark+1];
    }

    private String getMark() {
        Random random = new Random();
        double number = random.nextInt(20) * 5. / 20;

        return String.format("%.1f", number);
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        ViewHolder(View view) {
            super(view);
            view.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (mListener != null)
                        mListener.onClick(getAdapterPosition());
                }
            });
        }
    }

}

File: app/src/main/java/com/nollpointer/dates/ui/practise/PractiseCardsAdapter.kt
package com.nollpointer.dates.ui.practise

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.nollpointer.dates.R
import java.util.*

/**
 * @author Onanov Aleksey (@onanov)
 */
class PractiseCardsAdapter(private val title_texts: Array<String>, private val subtitle_texts: Array<String>, private val imageIds: IntArray, private val backgrounds: IntArray) : RecyclerView.Adapter<PractiseCardsAdapter.ViewHolder>() {
    private lateinit var marks: IntArray
    private var mListener: Listener? = null
    private var currentMode = 0

    private val mark: String
        get() {
            val random = Random()
            val number = random.nextInt(20) * 5.0 / 20
            return String.format("%.1f", number)
        }

    interface Listener {
        fun onClick(position: Int)
    }

    fun setListener(listener: Listener?) {
        mListener = listener
    }

    fun setMode(practiseMode: Int) {
        currentMode = practiseMode
    }

    fun setMarks(marks: IntArray) {
        this.marks = marks
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_practise, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val cardView = holder.itemView
        val titleTextView = cardView.findViewById<TextView>(R.id.info_text_title)
        val subTitleTextView = cardView.findViewById<TextView>(R.id.info_text_subtitle)
        //val markImageView = cardView.findViewById<ImageView>(R.id.info_image_mark)
        val imageView = cardView.findViewById<ImageView>(R.id.info_image)
        titleTextView.text = title_texts[position]
        subTitleTextView.text = subtitle_texts[position]
        imageView.setImageResource(imageIds[position])
        imageView.setBackgroundResource(backgrounds[position])
        imageView.contentDescription = title_texts[position]
       // if (currentMode == 1) {
           // markImageView.visibility = View.VISIBLE
           // markImageView.setImageResource(getMarkImage(marks[position]))
        //}
    }

    override fun getItemCount(): Int {
        return imageIds.size
    }

    private fun getMarkImage(mark: Int): Int { //int colors[] = {0xFFB71C1C,0xFFFFEB3B, 0xFF43a047};
        val colors = intArrayOf(R.drawable.ic_practise_result_nothing, R.drawable.ic_practise_result_bad, R.drawable.ic_practise_result_neutral, R.drawable.ic_practise_result_good)
        return colors[mark + 1]
    }

    inner class ViewHolder internal constructor(view: View) : RecyclerView.ViewHolder(view) {
        init {
            view.setOnClickListener { if (mListener != null) mListener!!.onClick(adapterPosition) }
        }
    }

}